src/alexapi/device_platforms/hyperion.py

```diff
@@ -0,0 +1,160 @@
+
+	def handle_indicate(self, mode, state=False, flash=False):
+		logger.debug("indicate %s %s", mode, self.display_state(state))
+		flash = self.should_flash(mode)
+		if not state and not flash:
+			self.hyperion_clear()
+		if state:
+			self.hyperion_indicate(self.get_color(mode), flash)
+
+	def hyperion_clear(self):
+		logger.debug("Clearing Hyperion settings")
+		self.hyperion_send(self.hyperion_message('clear', True))
+
+	def hyperion_effect(self, color, flash=False):
+		effect = {'args': {'color': color}}
+		if flash:
+			effect['name'] = "Strobe white"
+			effect['args']['frequency'] = self._pconfig['flash_frequency']
+		else:
+			effect['name'] = "Knight rider"
+			effect['args']['speed'] = self._pconfig['hyperion_effect_speed']
+		return {'effect': effect}
+
+	def hyperion_indicate(self, color, flash=False, duration=False):
+		command = self._pconfig['hyperion_mode']
+		if flash and command == 'color':
+			command = 'effect'
+		if flash:
+			duration = self._pconfig['flash_duration']
+		options = self.hyperion_options(command, color, duration, flash)
+		if self.setup_complete:
+			self.hyperion_send(self.hyperion_message(command, True, options))
+
+	def hyperion_options(self, command, color, duration=False, flash=False):
+		options = {'color': color}
+		if command == 'effect':
+			options = self.hyperion_effect(color, flash)
+		if duration:
+			options['duration'] = duration
+		return options
+
+	def hyperion_message(self, command, priority=False, options=None):
+		message = options or {}
+		message['command'] = command
+		if priority:
+			message['priority'] = self._pconfig['hyperion_priority']
+		return message
+
+	def hyperion_send(self, message):
+		if self._pconfig['verbose']:
+			logger.debug("sending '%s'", json.dumps(message))
+		if self.socket_status():
+			self.socket.send(json.dumps(message))
+		else:
+			logger.warning("Unable to send Hyperion state update")
+			self.init_connection()
+
+	def indicate_failure(self):
+		pass
+
+	def indicate_playback(self, state=True):
+		if self._pconfig['indicate_playback']:
+			self.handle_indicate('playback', state)
+
+	def indicate_processing(self, state=True):
+		self.handle_indicate('processing', state)
+
+	def indicate_recording(self, state=True):
+		self.handle_indicate('recording', state)
+
+	def indicate_success(self):
+		pass
+
+	def init_connection(self):
+		logger.debug('Initializing connection')
+		if self._pconfig['verbose']:
+			websocket.enableTrace(True)
+		self.socket = websocket.WebSocketApp(self.service,
+				on_message=self.on_socket_message,
+				on_close=self.on_socket_close,
+				on_error=self.on_socket_error)
+		self.socket_thread = threading.Thread(target=self.socket.run_forever)
+		self.socket_thread.daemon = True
+		self.socket_thread.start()
+
+	def on_socket_close(self, ws): # pylint: disable=unused-argument
+		logger.debug('Closing connection')
+
+	def on_socket_error(self, ws, error): # pylint: disable=unused-argument
+		logger.debug(error)
```

Comment:
There is also a connection check at startup and a
warning when AlexaPi is unable to send the the request when connection is
lost. So used debug here because even if there is a short error
(restarting your Hyperion instance or have the light turned of for some
time) it does not stop AlexaPi itself from working and i thought it would be annoying when you have a log full of errors when you turn your your light off over night. But could be changed to
an error.